src/controller/CHIPCommissionableNodeController.cpp

```diff
@@ -35,5 +37,39 @@ const Mdns::DiscoveredNodeData * CommissionableNodeController::GetDiscoveredComm
     return GetDiscoveredNode(idx);
 }
 
+CHIP_ERROR CommissionableNodeController::RequestCommissioning(int commissionerNumber)
+{
+#if CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY_CLIENT
+    const Mdns::DiscoveredNodeData * commissioner;
+    for (int i = 0; i < CHIP_DEVICE_CONFIG_MAX_DISCOVERED_NODES; i++)
+    {
+        commissioner = GetDiscoveredCommissioner(i);
+        if (commissioner != nullptr && --commissionerNumber == 0)
+        {
+            break;
+        }
+    }
+    if (commissionerNumber != 0)
```

Comment:
This does not seem correct to me.

Let's say we have discovered absolutely nothing and the caller passes in `0` as `commissionerNumber`.  We'll happily proceed past here and proceed to crash on a null-deref.

Reply:
If we discover absolutely nothing, we will only display that 0 commissioners were found and exit (see line #80 VerifyOrExit)

Reply:
My point is that this function's safety depends on something non-obvious being done by its callers.  Nothing documents that you MUST NOT call this function if nothing was discovered and that it will crash if you do.  The use right now happens to be safe, but it's way too easy for the caller to change and become unsafe.

It would be better if this function simply didn't do unsafe things no matter what the caller passed in and under what conditions.